concept/tests/test_statistics.rs

```diff
@@ -289,13 +289,15 @@ fn put_has_twice() {
     thing_manager.finalise(&mut snapshot).unwrap();
     snapshot.commit().unwrap().unwrap();
 
+    let mut synchronised = Statistics::new(SequenceNumber::MIN);
+    synchronised.may_synchronise(&storage).unwrap();
+
     let mut snapshot = storage.clone().open_snapshot_write_at(create_commit_seq);
     person.set_has_unordered(&mut snapshot, &thing_manager, name.as_reference()).unwrap();
     thing_manager.finalise(&mut snapshot).unwrap();
-    snapshot.commit().unwrap().unwrap();
+    snapshot.commit().unwrap_err(); // Can't concurrently modify the same 'has'
 
-    let mut synchronised = Statistics::new(SequenceNumber::MIN);
-    synchronised.may_synchronise(&storage).unwrap();
+    synchronised.sequence_number = synchronised.sequence_number + 1;
```

Comment:
There was another snapshot trying to commit, so it should be correct.

Reply:
does it even matter?

Reply:
I think you should move the statistics assert check to right after the new `may_synchronise()`, and then you can do the commit that fails, and then check the statistics has _not_ changed.